presidio-image-redactor/presidio_image_redactor/dicom_image_redactor_engine.py

```diff
@@ -733,12 +737,74 @@ def _add_redact_box(
 
         return redacted_instance
 
+    def _get_analyzer_results(
+        self,
+        image: PIL.PngImagePlugin.PngImageFile,
+        instance: pydicom.dataset.FileDataset,
+        redact_approach: Union[str, PatternRecognizer],
+        ocr_kwargs: Optional[dict],
+        **text_analyzer_kwargs
+    ) -> List[ImageRecognizerResult]:
+        """Analyze image with selected redaction approach.
+
+        :param image: DICOM pixel data as PIL image.
+        :param instance: DICOM instance (with metadata).
+        :param redact_approach: What approach to use when redacting
+        ("default", "metadata", "allow", or a PatternRecognizer object).
+        :param ocr_kwargs: Additional params for OCR methods.
+        :param text_analyzer_kwargs: Additional values for the analyze method
+        in AnalyzerEngine (e.g., allow_list).
+
+        :return: Analyzer results.
+        """
+        # Detect PII
+        if type(redact_approach) == str:
```

Comment:
I would consider using interfaces for a more opn-close design. next approach will just accumulate here and further lengthen the method (which is already complex)

Reply:
Also what if I want to use multiple approaches like before - metadata and text analyzer?

Reply:
I have made changes to match Omri's suggestion of using a `use_metadata=` bool arg instead and allowing for `ad_hoc_recognizers` to be passed in and used along with metadata redaction.

Resolving this comment as the original feedback here is no longer relevant with the code change.